src/Vite.AspNetCore/TagHelpers/ViteTagHelper.cs

```diff
@@ -112,10 +112,25 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
 			// If the Vite script was not inserted, it will be prepended to the current element tag.
 			if (ViteStatusService.IsMiddlewareRegistered && !this._status.IsDevScriptInserted)
 			{
+				var baseUrl = CombineUri(this._viteOptions.GetViteDevServerUrl(), this._viteOptions.Base ?? string.Empty);
 				// Build the uri to the vite client script.
-				var viteClientUrl = this._viteOptions.GetViteDevServerUrl() + "/@vite/client";
+				var viteClientUrl = baseUrl + "@vite/client";
 				// Add the script tag to the output
 				output.PreElement.AppendHtml($"<script type=\"module\" src=\"{viteClientUrl}\"></script>");
+
+				if (this._viteOptions.UseReactRefresh)
```

Comment:
About this. The [documentation](https://vitejs.dev/guide/backend-integration.html) says that the plugin-react script should be added before the other scripts.